src/codegen/llvm/codegen_llvm_visitor.cpp

```diff
@@ -753,19 +668,27 @@ void CodegenLLVMVisitor::visit_codegen_function(const ast::CodegenFunction& node
     block->accept(v);
 
     // Allocate parameters on the stack and add them to the symbol table.
-    ir_builder.allocate_function_arguments(func, arguments);
+    if (wrap_kernel_functions && node.get_is_kernel()) {
```

Comment:
Good catch - fixed that. Re explicit bitcasts, I would say it is maybe better to do it explicitly since if you generate IR for non-execution, you want to see the right type, etc.